test/types/queries.test.ts

```diff
@@ -311,16 +316,18 @@ function gh11964() {
 
   type WithId<T extends object> = T & { id: string };
 
-  class Repository<T extends object> {
-    /* ... */
+  type TestUser = {
```

Comment:
Please leave the `class Repository<>` here, this test case was specifically for an issue that occurred with generic classes.

Reply:
I actually did it because it was failing, but I figured out a way to fix this now by writing it like this and keeping the class Repository<>:
```
class Repository<T extends object> {
    find(id: string) {
      const idCondition: Condition<WithId<T>['id']> = id as Condition<WithId<T>['id']>;
      const filter: FilterQuery<WithId<T>> = { id } as FilterQuery<WithId<T>>;
    }
  }
```

Not sure if this is the best way, but its passing the tests + you still get autocompletion for stuffs like `$eq`, `$in`, etc. in the above test too

Reply:
I would strongly prefer if this test passed as written before we merge this PR. I won't say it's a hard blocker, but we shouldn't merge this PR unless either this test passes or we have a very good explanation of why this test is failing. And "just write code this other way" is generally not a good enough reason, because different people write code in different ways and we should meet users where they are.

Reply:
That is the reason why I wrote the above test. If you approve it, I can write the above test in a separate test function and push that commit to this PR. As for the existing old test, do suggest what should be done about it?

Reply:
@vkarpov15 Also, this is a wrong statement which will always fail even if you give a concrete interface implementation instead of just parameter `T`:
```
const idCondition: Condition<WithId<T>>['id'] = id;
```

The reason is because `Condition<K>` will return something like `K | K[] | ...`. Now, if the return type of `Condition<K>` is `K[]`, then it will not have the property `['id']` obviously since its an array. So, the above statement was also written with incorrect typescript assumption

Reply:
@vkarpov15 I have added new commit separating out the tests. Can you review the changes now?